addons/xm157/functions/fnc_keyPress.sqf

```diff
@@ -22,6 +22,11 @@ if (!GVAR(shown)) exitWith { false }; // fast exit if not shown
 if (!([ACE_player, objNull, ["isNotInside"]] call EFUNC(common,canInteractWith))) exitWith { false };
 if (!(ACE_player call CBA_fnc_canUseWeapon)) exitWith { false };
 
+private _scope = (ACE_player weaponAccessories currentMuzzle ACE_player) param [2, ""];
+
+if (_scope == "") exitWith { false }; // should not happen, but put as a precaution
+if (getText (_scope call CBA_fnc_getItemConfig >> "weaponInfoType") != QGVAR(info)) exitWith { false }; // allow zeroing changes for muzzles not using a scope using the XM157 framework
+
```

Comment:
I don't think `weaponAccessories` supports muzzles so it's probably just easier to do
```suggestion
if (currentMuzzle ACE_player != currentWeapon ACE_player) exitWith { false };
```

Reply:
> I don't think `weaponAccessories` supports muzzles so it's probably just easier to do

It reports `["", "", "", ""]` for the vanilla GL, so I assumed it works. But thinking about it a bit harder, it probably doesn't.

If we implement your suggestion, it means that non-primary muzzles cannot have XM157-framework based optics. Is that your intention?

Reply:
I think that's fine
I don't think any GLs use the optic sights anyways
and if they could I don't think you have enough mil range in the sight (you need extreme angles to launch)
The only useful case might be that apex 5.56/50 cal combo rifle thing but that's pretty obscure

Reply:
I'll check the combo rifle and see if it could a problem. If not, I'll commit your suggestion.

Reply:
Combo rifle (classname `arifle_ARX_blk_F`) would be able to benefit from having the XM157 feature available for both muzzles. However, my proposed solution does not work.

But even more importantly:  The ballistics computer does not support secondary muzzles, so your solution is fine.